tasks/eva_2038/storeSSIDsInVariantWarehouse.py

```diff
@@ -0,0 +1,114 @@
+log_cfg.add_stdout_handler()
+log_cfg.set_log_level(logging.INFO)
+
+
+def run_nextflow(working_dir, params, project_id, accession_report_file, db_name):
+    random_string = ''.join(random.choice(string.ascii_letters) for i in range(6))
+    workflow_name = f"import_accession_{project_id}_{random_string}"
+
+    # create nextflow work dir - remove if already exists
+    nextflow_work_dir = os.path.join(working_dir, project_id, workflow_name)
+    if os.path.exists(nextflow_work_dir):
+        shutil.rmtree(nextflow_work_dir)
+    os.makedirs(nextflow_work_dir)
+
+    try:
+        command_utils.run_command_with_output(
+            f'Nextflow {workflow_name} process',
+            ' '.join((
+                'export NXF_OPTS="-Xms1g -Xmx8g"; ',
+                params['nextflow_path'], params['nextflow_script'],
+                '-work-dir', nextflow_work_dir,
+                '--java_app', params['java_app'],
+                '--acc_import_job_props', params['acc_import_job_props'],
+                '--accession_report', accession_report_file,
+                '--db_name', db_name,
+                '--logs_dir', os.path.join(working_dir, project_id)
+            ))
+        )
+    except subprocess.CalledProcessError as e:
+        logger.error(f'Nextflow {workflow_name} pipeline failed: results might not be complete.')
+        raise e
+
+
+def get_db_name(params, project_id, accession_report_file):
+    with get_metadata_connection_handle("production_processing", params['private_settings_xml_file']) as pg_conn:
+        query = f"""
+            select distinct asm.assembly_accession, t.taxonomy_id
+            from project_taxonomy pt 
+            join taxonomy t on t.taxonomy_id = pt.taxonomy_id 
+            join project_analysis pa on pa.project_accession = pt.project_accession 
+            join analysis a on pa.analysis_accession = a.analysis_accession 
+            join assembly asm on asm.assembly_accession = a.vcf_reference_accession 
+            join analysis_file af on af.analysis_accession = pa.analysis_accession 
+            join file f on f.file_id = af.file_id 
+            where pt.project_accession = '{project_id}' 
+            and f.filename = '{os.path.basename(accession_report_file).replace(".accessioned.vcf", ".vcf")}'
+            and f.file_type IN ('vcf', 'VCF', 'vcf_aggregate', 'VCF_AGGREGATE')
+            """
+        res = list(get_all_results_for_query(pg_conn, query))
+        if res:
+            return resolve_variant_warehouse_db_name(pg_conn, res[0][0], res[0][1])
+        else:
+            return None
+
+
+def run_import_accession_job_for_project(working_dir, params, project_id):
+    logger.info(f"Starting processing project: {project_id}")
+
+    project_path = os.path.join(params['project_dir'], project_id)
+    if not os.path.exists(project_path):
+        logger.error(f"Could not find the Project {project_id}. Please make sure the project {project_path} exists")
```

Comment:
For older studies, I guess we'll use [this script](https://github.com/EBIvariation/eva-submission/blob/61f4da73214e871a4a98cbe4f2ce5c164d09debe/bin/retrieve_archives_from_lts.py) to fetch these from LTS?

I wonder if it's worth including this and the very first step (determining the project IDs) in the automation, rather than just the accession load as you've done here... perhaps it's easier to manage these manually though.

Reply:
I think it's worth adding something that looks in the FTP. This will have to be done from a datamover node so you might have to run this script from datamover or retrieve the accession via a hard coded srun command.